R/ncdf.R

```diff
@@ -574,9 +678,9 @@ read_ncdf = function(.x, ..., var = NULL, ncsub = NULL, curvilinear = character(
         t_dim = NA_character_
       # https://github.com/r-spatial/stars/issues/378
       if (!is.na(t_dim) && names(coords)[i] == t_dim)
-        dimensions[[i]]$offset[1L] = coords[[i]][dims$start[i]]
+        dimensions[[i]]$offset[1L] = coords[[i]][1]
       else
-        dimensions[[i]]$offset[1L] = coords[[i]][dims$start[i]] - mdc/2
+        dimensions[[i]]$offset[1L] = coords[[i]][1] - mdc/2
```

Comment:
This was a bug -- dims$start[i] is usually 1 but not always. We always want the first element of the coords list element to figure out the offset.